packages/components/textarea/src/utils/helper-functions.js

```diff
@@ -0,0 +1,11 @@
+import { NEW_LINE_CHARACTER, ROWS_MAX, ROWS_MIN } from './constants';
+
+export function getRowsInRange(value) {
+	if (value <= ROWS_MIN) return ROWS_MIN;
+	if (value >= ROWS_MAX) return ROWS_MAX;
+	return value;
+}
+
+export function countLines(text) {
+	return text.split(NEW_LINE_CHARACTER).length;
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Make sure that using a regular expression is safe here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS4784)